### DISCUSSION TITLE
Курсы валют

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#297] [Как это сделать?] [answered]
Всем добрый день. В общем тема следующая: есть потребность в получении данных с какого либо российского сервиса в виде актуальных курсов валют. Актуальная цена валюты по отношению к рублю должна подтягиваться через API по нажатию на кнопку.  Самый банальный и вроде бы стабильный российский сервис это центробанк (https://www.cbr.ru/development/dws/), но дату он способен передать только в вид xml и к тому же все через протокол SOAP. Вот теперь вопрос: как правильно разобрать xml в тотуме и вообще каким образом можно получить ответ по данному API от цб?

**Comment by juramebel** [comment]
Здравствуйте!
Функция xmlExtract не помогает?
Можете поделиться запросом, по которому получаете xml?

**Comment by juramebel** [comment, marked as answer]
Настроил у себя через этот ресурс:

usd:$extr[Valute][USD][Value]
eur:$extr[Valute][EUR][Value]

extr: jsonExtract(text: $getfs)

getfs: getFromScript(uri: "https://www.cbr-xml-daily.ru/daily_json.js")

**Reply by Vadimgrig2012** [reply]
Спасибо вам. Уже через другой сервис по API настроился, но ваш вариант тоже будет очень кстати...
